app.py

```diff
@@ -502,75 +356,16 @@ def create_redirect():
     return flask.redirect('https://docs.snapcraft.io/build-snaps')
 
 
-def convert_limit_offset_to_size_page(link):
-    url_parsed = urlparse(link)
-    host_url = (
-        "{base_url}"
-        "?q={q}&limit={limit}&offset={offset}"
-    )
-
-    url_queries = parse_qs(url_parsed.query)
-    q = url_queries['q'][0]
-    size = int(url_queries['size'][0])
-    page = int(url_queries['page'][0])
-
-    return host_url.format(
-        base_url=flask.request.base_url,
-        q=q,
-        limit=size,
-        offset=size*(page-1)
-    )
-
-
-def get_pages_details(links):
-
-    links_result = {}
-
-    if('first' in links):
-        links_result['first'] = convert_limit_offset_to_size_page(
-            links['first']['href']
-        )
-
-    if('last' in links):
-        links_result['last'] = convert_limit_offset_to_size_page(
-            links['last']['href']
-        )
-
-    if('next' in links):
-        links_result['next'] = convert_limit_offset_to_size_page(
-            links['next']['href']
-        )
-
-    if('prev' in links):
-        links_result['prev'] = convert_limit_offset_to_size_page(
-            links['prev']['href']
-        )
-
-    if('self' in links):
-        links_result['self'] = convert_limit_offset_to_size_page(
-            links['self']['href']
-        )
-
-    return links_result
-
-
 # Normal views
 # ===
 @app.route('/')
```

Comment:
If we're splitting the methods into separate files I feel we should put the route definitions there too... I find it weird to declare the route in one file but implement the behaviour in another. Maybe it's just me?

Reply:
After a discussion we had with @nottrobin we decided to keep the routes in the "entrance" file to keep the backend simple. So that if a new developer arrives on the project he can open the `app.py` file and follow the path of the issue in a simple way

Reply:
I don't like it, but I'm okay with it.